mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { FormTokenField as WpFormTokenField } from '@wordpress/components';
+import { uniq } from 'lodash';
+
+export type FormTokenItem = {
+  id: number | string;
```

Comment:
Can't we convert it and use just one of those types?

Reply:
user roles IDs are strings while list segment IDs are integers. I preferred to mark this distinctively. The diff to change that would look like this:
```
diff --git a/mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx b/mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx
index edb20d8fc..64341c2a1 100644
--- a/mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx
+++ b/mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx
@@ -1,7 +1,7 @@
 import { FormTokenField as WpFormTokenField } from '@wordpress/components';
 
 export type FormTokenItem = {
-  id: number | string;
+  id: string;
   name: string;
 };
 
diff --git a/mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx b/mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx
index eb41919d1..27ae805c2 100644
--- a/mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx
+++ b/mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx
@@ -15,14 +15,14 @@ export function ListPanel(): JSX.Element {
   );
 
   const rawSelected = selectedStep.args?.segment_ids
-    ? (selectedStep.args.segment_ids as number[])
+    ? (selectedStep.args.segment_ids as string)
     : [];
 
   const validSegments = segments.filter(
     (segment) => segment.type === 'default',
   );
   const selected = validSegments.filter((segment): boolean =>
-    rawSelected.includes(segment.id as number),
+    rawSelected.includes(segment.id),
   );
   return (
     <PanelBody opened>
david@pc:~/projects/mailpoet/mailpoet$ git checkout .
Updated 2 paths from the index
david@pc:~/projects/mailpoet/mailpoet$ git diff
diff --git a/mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx b/mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx
index edb20d8fc..64341c2a1 100644
--- a/mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx
+++ b/mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx
@@ -1,7 +1,7 @@
 import { FormTokenField as WpFormTokenField } from '@wordpress/components';
 
 export type FormTokenItem = {
-  id: number | string;
+  id: string;
   name: string;
 };
 
diff --git a/mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx b/mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx
index eb41919d1..168032499 100644
--- a/mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx
+++ b/mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx
@@ -15,14 +15,14 @@ export function ListPanel(): JSX.Element {
   );
 
   const rawSelected = selectedStep.args?.segment_ids
-    ? (selectedStep.args.segment_ids as number[])
+    ? (selectedStep.args.segment_ids as string[])
     : [];
 
   const validSegments = segments.filter(
     (segment) => segment.type === 'default',
   );
   const selected = validSegments.filter((segment): boolean =>
-    rawSelected.includes(segment.id as number),
+    rawSelected.includes(segment.id),
   );
   return (
     <PanelBody opened>
```
We would pretend the segment ID is a string, while it is not. JavaScript will still send an integer to the Rest API (so we are "typesafe" in PHP), but I preferred to spell out the real types in TypeScript and it feels to me otherwise, we would obscure that fact.

What do you think?